longest-substring-without-repeating-characters/sunjae95.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @description
+ * brainstorming:
+ * hash table + two pointer
+ *
+ * n = length of s
+ * time complexity: O(n)
+ * space complexity: O(n)
+ */
+var lengthOfLongestSubstring = function (s) {
+  const map = new Map();
+  let answer = 0;
+  let start = 0;
+  let end = 0;
+  for (let i = 0; i < s.length; i++) {
+    if (map.has(s[i])) start = Math.max(map.get(s[i]) + 1, start);
+    map.set(s[i], i);
+    end += 1;
+    answer = Math.max(answer, end - start);
```

Comment:
굳이 `end`가 필요한가? 생각이 들었습니다.

```suggestion
  // let end = 0;
  for (let i = 0; i < s.length; i++) {
    if (map.has(s[i])) start = Math.max(map.get(s[i]) + 1, start);
    map.set(s[i], i);
    // end += 1;
    answer = Math.max(answer, i - start + 1);
```

Reply:
오호.. end가 i와 동작을 동일하게 하는건 작성하면서 못알아챘네요.
제가 지금봐도 리뷰해주신 방법이 더 좋은거 같아요!👍